lib/hci-socket/gatt.js

```diff
@@ -536,7 +536,10 @@ Gatt.prototype.write = function(serviceUuid, characteristicUuid, data, withoutRe
       var opcode = data[0];
 
       if (opcode === ATT_OP_WRITE_RESP) {
-        this.emit('write', this._address, serviceUuid, characteristicUuid);
+        this.emit('write', this._address, serviceUuid, characteristicUuid, null);
+      }
+      else if(opcode === ATT_OP_ERROR){
+        this.emit('write', this._address, serviceUuid, characteristicUuid, "ATT_OP_ERROR - error code : " + resp[4]);
```

Comment:
You could fork @Lahorde's fork, change it to `data`, and then PR into `Lahorde:fork_branch` which would make its way into here once merged.